boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.dts

```diff
@@ -156,9 +156,28 @@ arduino_serial: &flexcomm12 {
 
 	pca9420: pca9420@61 {
 		reg = <0x61>;
-
+		compatible = "regulator-pmic";
+		/*
+		 * This mode-reg setting will permit mode control by
+		 * pins, a feature of the PCA9240. If mode control via
+		 * I2C is desired, the following settings
+		 * should be used:
+		 * regulator-initial-mode = <PCA9420_MODECFG0_I2C>;
+		 * regulator-allowed-modes = <PCA9420_MODECFG0_I2C>,
+		 *			<PCA9420_MODECFG1_I2C>,
+		 *			<PCA9420_MODECFG2_I2C>,
+		 *			<PCA9420_MODECFG3_I2C>;
+		 * modesel-reg = <PCA9420_TOP_CNTL3>;
+		 * modesel-mask = <PCA9420_TOP_CNTL3_MODE_I2C_MASK>;
+		 */
+		regulator-allowed-modes = <PCA9420_MODECFG0_PIN>,
+					<PCA9420_MODECFG1_PIN>,
+					<PCA9420_MODECFG2_PIN>,
+					<PCA9420_MODECFG3_PIN>;
+		regulator-initial-mode = <PCA9420_MODECFG0_PIN>;
+		modesel-reg = <PCA9420_MODECFG_0_0>;
```

Comment:
Should there be a `modesel-reg` per mode

Reply:
The `modesel-reg` will be offset based on which mode is selected, so we only need to encode the first `modesel-reg` value in DTS